### PR TITLE
Warn if dynamic template contains non-escaped character

### LINKED ISSUE #790: Document the difference of using three brackets vs two brackets with a template and throw or log an error when used improperly
#### Issue Summary

Our templates use the handlebars.js syntax as documented [here](https://sendgrid.com/docs/ui/sending-email/using-handlebars/). When including the characters ', " or & in a subject line replacement you need to use three brackets vs two ( {{{ content }}} vs {{ content }} ). 

#### Acceptance Criteria

* Documentation of this functionality in the use cases section (transactional templates) of this repo.
* A new test is added that demonstrates that if you use the characters ', " or & in a subject line replacement utilizing template id's prefixed with d-, the code will throw an error or log (to avoid a breaking change) the incident.

### PR DESCRIPTION
**Pull request by reedsa**
# Closes #790 

### Checklist
- [X] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [X] I have read the [Contribution Guide] and my PR follows them.
- [X] I updated my branch with the master branch.
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation about the functionality in the appropriate .md file
- [X] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Warns if any of the ', " or & characters are used in a dynamic template that does not escape with triple-stash syntax

If you have questions, please send an email to [Sendgrid](mailto:dx@sendgrid.com), or file a Github Issue in this repository.

### CODE REVIEW

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

File: packages/helpers/classes/mail.spec.js [hunk 1]
```diff
@@ -154,4 +155,63 @@ describe('Mail', function() {
     });
 
   });
+
+  describe('dynamic template handlebars substitutions', () => {
```
Comment (Ditofry) [inline packages/helpers/classes/mail.spec.js]: righteous

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br /><br />                                         We noticed that you have earned 3 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold.<br /><br />                                         Also, please be sure that you have officially registered <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 6 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid

**Comment by umarhussain15** [comment]
Any option to stop this warning message other than escaping characters? 
Firebase shows this warning as an error in logs which makes it difficult to monitor logs for debugging.

Thanks.

**Comment by thinkingserious** [comment]
Hello @umarhussain15,

Would you mind creating a new ticket for your request? That way we can properly track progress. Thanks!

With best regards,

Elmer

**Comment by umarhussain15** [comment]
Hi @thinkingserious I just checked latest version of release and https://github.com/sendgrid/sendgrid-nodejs/pull/932 has added a solution to hide warnings for dynamic template data. So I think there is no need to create a new ticket.
Thanks for the response.

**Comment by thinkingserious** [comment]
Awesome, thanks for the follow up!

**Comment by prBigBrother** [comment]
Hi @thinkingserious! 

can you push forward this pull request with new release? 
Right now it's makes really difficult to read logs because our app sending too much emails.

Thank you a lot

**Comment by noway** [comment]
How do I get rid of that warning? I have 3 curly brackets, why is it littering my logs? There's no way to disable it? I need my logs clean for logging purposes.

Cheers

File: packages/helpers/constants/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const DYNAMIC_TEMPLATE_CHAR_WARNING = `
+Content with characters ', " or & may need to be escaped with three brackets
+{{{ content }}}
+See https://sendgrid.com/docs/ui/sending-email/using-handlebars/ for more information.`;
```
Comment (ekeklol) [inline packages/helpers/constants/index.js]: url leads to -> 404 not found page

Comment (childish-sambino) [reply packages/helpers/constants/index.js]: Fixed by https://github.com/sendgrid/sendgrid-nodejs/pull/943

**Pull request merged**